### PR TITLE
feat: macos support

### PR DESCRIPTION
**Pull request by okwasniewski**
This PR solves #46 

## Unresolved issue

The view is not sized properly when using sidebar, it's width is always width of the window.

CleanShot 2024-10-19 at 20 32 34@2x

### CODE REVIEW

**Comment by okwasniewski** [comment]
cc: @Saadnajmi maybe you know what I've missed to make it work with the sidebar? 

I've been also getting these kind of errors, not sure if this is related to RN macOS or my implementation: 

CleanShot 2024-10-19 at 20 46 36@2x

File: ios/TabViewProvider.swift [hunk 1]
```diff
@@ -154,12 +175,23 @@ struct TabData: Codable {
...
+#elseif os(macOS)
+    let newImage = NSImage(size: size)
+    newImage.lockFocus()
+    self.draw(in: CGRect(origin: .zero, size: size),
+              from: CGRect(origin: .zero, size: self.size),
+              operation: .copy,
+              fraction: 1.0)
+    newImage.unlockFocus()
+    return newImage
+#endif
```
Comment (Saadnajmi) [inline ios/TabViewProvider.swift:195]: See a similar change here: https://github.com/microsoft/react-native-macos/pull/2209
`lockFocus` and that pattern is deprecated in favor of [imageWithSize:flipped:drawingHandler:](https://developer.apple.com/documentation/appkit/nsimage/1519860-imagewithsize). Similarly, I think `UIGraphicsBeginImageContextWithOptions` is mostly replaced with `UIGraphicsImageRenderer`

Comment (okwasniewski) [reply ios/TabViewProvider.swift:195]: Thanks! I will fix it

File: apps/example/macos/Podfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  File.exist?("#{ws_dir}/node_modules/react-native-test-app/macos/test_app.rb") ||
+  ws_dir.expand_path.to_s == '/'
+require "#{ws_dir}/node_modules/react-native-test-app/macos/test_app.rb"
+
+workspace 'ReactNativeBottomTabs.xcworkspace'
+
+use_test_app! do |test_app|
+  # Workaround for not using use_frameworks! in the Podfile
+  pod 'SDWebImage', :modular_headers => true
+  pod 'SDWebImageSVGCoder', :modular_headers => true
```
Comment (Saadnajmi) [inline apps/example/macos/Podfile:12]: Are these because expo-image uses them?

Comment (okwasniewski) [reply apps/example/macos/Podfile:12]: Bottom tabs use them, it's a workaround for not needing to use static_frameworks - in the installation instructions I recommend to use static frameworks (or people could also enable modular headers globally but I had some issues with it). 

It's about cocoapods and mixing Swift with Objective-C

**Pull request merged**